cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -80,20 +81,55 @@
   private final String projectId;
   private final String network;
   private final String zone;
-  private final String systemNetwork;
+  private final boolean useInternalIP;
 
   public static DataprocClient fromConf(DataprocConf conf) throws IOException, GeneralSecurityException {
+    ClusterControllerClient client = getClusterControllerClient(conf);
+    Compute compute = getCompute(conf);
+
+    String network = conf.getNetwork();
     String systemNetwork = null;
     try {
       systemNetwork = DataprocConf.getSystemNetwork();
-    } catch (IllegalArgumentException e) {
-      // expected when not running on GCP
+    } catch (Exception e) {
```

Comment:
Is the `DataprocConf.getSystemNetwork` method throwing something other than IllegalArgumentException?
If not - wouldn't it be better to handle the specific exception thrown?

Reply:
hm not sure why I changed this, think I can change it back